### PR TITLE
Retryable writes prerequisites

### PR DESCRIPTION
**Pull request by p-mongo**
This PR contains changes that were made as part of retryable writes work but aren't part of the ticket proper, to avoid making the next PR unreadable.

https://jira.mongodb.org/browse/RUBY-1333

### CODE REVIEW

File: spec/mongo/cluster/topology_spec.rb [hunk 1]
```diff
@@ -105,28 +105,6 @@
           expect(topology).to be_a(Mongo::Cluster::Topology::Unknown)
         end
       end
-
```
Comment (saghm) [inline spec/mongo/cluster/topology_spec.rb]: What's the rationale for removing these tests?

File: spec/mongo/retryable_writes_spec.rb [hunk 2]
```diff
@@ -1,754 +1,751 @@
 require 'spec_helper'
 
-describe 'Retryable Writes' do
```
Comment (saghm) [inline spec/mongo/retryable_writes_spec.rb]: Wow, Github's calculated diff of this file is _terrible_...

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -25,22 +39,40 @@
 # CLIENT_DEBUG: Show debug messages from the client.
 #   CLIENT_DEBUG=1
 #
-# MONGODB_URI: Connection string to use instead of 127.0.0.1:27017.
-# Specify RS_ENABLED or SHARDED_ENABLED if connecting to a replica set
-# or a sharded cluster, otherwise the test suite will establish a
-# direct connection to the first host in the URI.
-#   MONGODB_URI=127.0.0.1:27001,127.0.0.1:27002
+# MONGODB_URI: Connection string to use. This must be a valid MongoDB URI;
```
Comment (saghm) [inline spec/spec_helper.rb]: I have to wonder if it's worth reducing the number of these variables at some point in the future; it seems to me that `MONGODB_URI` and `TOPOLOGY` should be sufficient. It's out of scope for this ticket though, so we can revisit another time.

File: spec/spec_tests/retryable_writes_spec.rb [hunk 4]
```diff
@@ -19,7 +19,7 @@
           let(:client) do
             authorized_client_with_retry_writes
           end
-
```
Comment (saghm) [inline spec/spec_tests/retryable_writes_spec.rb]: I've noticed a lot of places with small formatting issues like this. After the 2.6.0 release, we should discuss standardizing on some code formatting tool.

File: spec/mongo/cluster/topology_spec.rb (hunk 1 above)
Comment (p-mongo) [reply spec/mongo/cluster/topology_spec.rb]: As mentioned in the commit these tests are duplicated.

File: spec/spec_helper.rb (hunk 3 above)
Comment (p-mongo) [reply spec/spec_helper.rb]: I intend to redo test suite configuration at a future time. There are several other considerations that need to be addressed, for example some tests connect to sockets directly which can't be expressed with either MONGODB_URI or MONGODB_ADDRESSES.

**Pull request merged**